modules/packages/LinearAlgebra.chpl

```diff
@@ -2829,15 +2829,38 @@ module Sparse {
     }
     // matrix-matrix
     else if Adom.rank == 2 && Bdom.rank == 2 {
-      if !isCSArr(A) || !isCSArr(B) then
-        compilerError("Only CSR format is supported for sparse multiplication");
-      return _csrmatmatMult(A, B);
+      if !isCSArr(A) || !isCSArr(B) then {
+        if isCSArr(A) && !isCSArr(B) then {
+          return sparseDenseMatmul(A, B);
+        }
+        else if !isCSArr(A) && isCSArr(B) then {
+          var res = sparseDenseMatmul(B.T, A.T);
+          return res.T;
+        }
+      }
+      else {
+        return _csrmatmatMult(A, B);
+      }
     }
     else {
       compilerError("Ranks are not 1 or 2");
     }
   }
 
+  // Method returns the product of a Sparse and a Dense Matrix.
+  private proc sparseDenseMatmul(A: [?ADom], B: [?BDom]) where isSparseArr(A) || !isSparseArr(B) {
+    if ADom.dim(1) != BDom.dim(0) then
+        halt("Mismatched shape in sparse-dense matrix multiplication");
+
+    var resDom = {0..<A.domain.shape(0), 0..<B.domain.shape(1)};
+    var C: [resDom] A.eltType;
+
+    for i in 0..<B.domain.shape(1) {
```

Comment:
should this be a `forall`? @lydia-duncan and @e-kayrakli might have an opinion here.

Reply:
It seems like it would be reasonable for this to be a `forall`, yeah - A and B aren't changed, and C only seems to be changed using the iteration variable as an index, so updates should be independent of each other.